### PR TITLE
[k8s]handling module status based on pod condition object

### PR DESCRIPTION
**Pull request by ggjjj**

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs [hunk 1]
```diff
@@ -83,19 +87,23 @@ static ReportedModuleStatus ConvertPodStatusToModuleStatus(Option<V1PodStatus> p
...
+                                        {
                                             return new ReportedModuleStatus(ModuleStatus.Stopped, $"Module Stopped reason: {c.State.Terminated.Reason}");
+                                        }            
                                     }
                                     else
                                     {
                                         return new ReportedModuleStatus(ModuleStatus.Backoff, $"Started at {c.State.Running.StartedAt}");
                                     }
-                                }).GetOrElse(() => new ReportedModuleStatus(ModuleStatus.Failed, $"Module Failed with container status Unknown More Info: K8s reason: {status.Reason} with message: {status.Message}"));
+                                }).GetOrElse(() => new ReportedModuleStatus(ModuleStatus.Failed, $"Module Failed with container status Unknown More Info: K8s reason: {status.Conditions.Last().Reason} with message: {status.Conditions.Last().Message}"));
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs]: Are you somehow guaranteeing order in your Enumerable here such that you can depend on .Last? Why not .First or if this Enumerable has only one item, Single? Also, IIRC, Last/Single/First will throw if the Enumerable is empty... is there a chance that it is empty? If so, I sugest you use the OrDefault versions of these functions.

Comment (kkdawkins) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs]: Comment applies to all uses of Last.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs [hunk 2]
```diff
@@ -61,15 +61,19 @@ static ReportedModuleStatus ConvertPodStatusToModuleStatus(Option<V1PodStatus> p
                                     else if (c.State.Terminated != null)
                                     {
                                         if (c.State.Terminated.ExitCode != 0)
+                                        {
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs]: Why the formatting change?

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesRuntimeInfoProviderTest.cs [hunk 3]
```diff
@@ -305,25 +319,25 @@ public static IEnumerable<object[]> GetListOfPodsInAbnormalPhase()
             {
                 new object[]
                 {
-                    CreatePodWithPodParametersOnly("Running", "Unschedulable", "persistentvolumeclaim module-a-pvc not found"),
+                    CreatePodWithPodParametersOnly("Pending", null, null, "Unschedulable", "persistentvolumeclaim module-a-pvc not found"),
```
Comment (kkdawkins) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesRuntimeInfoProviderTest.cs]: Null or string.Empty? The latter is generally safer.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs (hunk 2 above)
Comment (ggjjj) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/PodExtensions.cs]: Rider suggestion of inconsistent brackets

**Comment by kkdawkins** [comment]
LGTM but I would get someone like @darobs who has Kubernetes knowledge to also look at it :)

**Pull request merged**